CMakeLists.txt

```diff
@@ -140,13 +146,21 @@ pkg_check_modules(SNDFILE REQUIRED "sndfile >= 1.0.25")
 add_includes_ldflags ("${SNDFILE_LDFLAGS}" "${SNDFILE_INCLUDE_DIRS}")
 
 ## Additional features ##
-
-# CppUnit
-option (ENABLE_CPPUNIT "Build CppUnit test instead of xournalpp application" OFF)
-if (ENABLE_CPPUNIT)
-	pkg_check_modules(CppUnit REQUIRED "cppunit >= 1.12-0")
+# Testing Framework: gtest
+option (ENABLE_GTEST "Enable gtest build for xournalpp application" OFF)
+if (ENABLE_GTEST)
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        URL https://github.com/google/googletest/archive/refs/tags/release-1.11.0.zip
+    )
+	# Prevent reloading if already downloaed
+	set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
+    # For Windows: Prevent overriding the parent project's compiler/linker settings
+    set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+    FetchContent_MakeAvailable(googletest)
 	enable_testing()
-endif (ENABLE_CPPUNIT)
+endif (ENABLE_GTEST)
```

Comment:
**Potential follow-up PR:** Doing the above only if [`find_package(GTest CONFIG)`](https://github.com/google/googletest/blob/master/googletest/README.md#incorporating-into-an-existing-cmake-project) fails might be nice. (I can try adding this!)